### PR TITLE
feat(onboarding): create OnboardingCache object

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- Use the onboardingCache object created in https://github.com/renovatebot/renovate/pull/20733 to store onboardingBranch related info like: `isModified` and `isConflicted` 

## Context
- https://github.com/renovatebot/renovate/issues/20797#issuecomment-1464903610
- It will reduce the number of git operations performed when processing existing onboardingPrs

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by RahulGautamSingh** [comment]
Blocked by https://github.com/renovatebot/renovate/pull/20893

File: lib/workers/repository/finalise/repository-statistics.ts [hunk 1]
```diff
@@ -57,6 +59,8 @@ function branchCacheToMetadata({
     branchSha,
     isModified,
     isPristine,
+    isConflicted,
+    isBehindBase,
```
Comment (rarkins) [inline lib/workers/repository/finalise/repository-statistics.ts:63]: Why the need to now log these in metadata?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/repository/index.ts [hunk 2]
```diff
@@ -73,6 +74,15 @@ export async function renovateRepository(
...
       addSplit('update');
       if (performExtract) {
+        if (!config.repoIsOnboarded) {
+          branches.push({
+            branchName: config.onboardingBranch!,
+            baseBranch: config.defaultBranch!,
+            automerge: false,
+            upgrades: [],
+            manager: '' as never, // onboarding branch won't have this field
+          } satisfies BranchConfig);
```
[the remark this answers is not in the document]
Comment (RahulGautamSingh) [reply lib/workers/repository/index.ts:84]: You mean like this?
```js
cache = {
scan: {master: {...}},
branches: [ {...}, {...}],
onboarding: {...}
}
```

File: lib/workers/repository/finalise/repository-statistics.ts (hunk 1 above)
Comment (RahulGautamSingh) [reply lib/workers/repository/finalise/repository-statistics.ts:63]: I added it for debugging, will remove once done.

**Comment by RahulGautamSingh** [comment]
Blocked by: https://github.com/renovatebot/renovate/pull/20733

**Pull request closed without merging**